internal/common/config.go

```diff
@@ -67,7 +69,24 @@ func DefaultConfig(role util.Role) *DolphinSchedulerConfig {
 	return &DolphinSchedulerConfig{
 		resources: resources,
 		common: &GeneralNodeConfig{
-			Affinity:                       &corev1.Affinity{}, // TODO
+			Affinity: &corev1.Affinity{
+				PodAntiAffinity: &corev1.PodAntiAffinity{
+					PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+						{
+							Weight: 70,
+							PodAffinityTerm: corev1.PodAffinityTerm{
+								TopologyKey: "kubernetes.io/hostname",
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										constants.LabelKubernetesInstance:  crName,
+										constants.LabelKubernetesComponent: string(role),
+									},
+								},
+							},
+						},
+					},
+				},
+			}, // TODO
```

Comment:
What is about `TODO`

Reply:
ok, i think we should rmove the affinity builder into the operator-go ?   @whg517